File: android/mekuri/src/main/kotlin/app/mekuri/MekuriTurnDecision.kt
package app.mekuri

/** Outcome of releasing a drag. */
internal enum class MekuriTurnDecision {
    Commit,
    Revert,

    ;

    internal companion object {
        /**
         * `velocity` is measured along the turn, in dp per second, and compared
         * signed: a fling against the turn never commits.
         */
        internal fun resolve(
            progress: Float,
            velocity: Float,
            configuration: MekuriConfiguration,
        ): MekuriTurnDecision =
            if (progress >= configuration.snapThreshold || velocity >= configuration.flingVelocity) {
                Commit
            } else {
                Revert
            }
    }
}

File: ios/Sources/Mekuri/MekuriTurnDecision.swift
import CoreGraphics

/// Outcome of releasing a drag.
enum MekuriTurnDecision: Equatable, Sendable {
    case commit
    case revert

    /// `velocity` is measured along the turn, in points per second; a fling
    /// against the turn never commits.
    static func resolve(progress: CGFloat, velocity: CGFloat, configuration: MekuriConfiguration) -> MekuriTurnDecision {
        progress >= configuration.snapThreshold || velocity >= configuration.flingVelocity ? .commit : .revert
    }
}
